crates/wasm-smith/src/core.rs

```diff
@@ -2494,15 +2497,21 @@ pub(crate) fn configured_valtypes(config: &Config) -> Vec<ValType> {
             // contain a non-null self-reference are also impossible to create).
             true,
         ] {
-            use AbstractHeapType::*;
-            for ty in [
-                Any, Eq, I31, Array, Struct, None, Func, NoFunc, Extern, NoExtern,
-            ] {
-                valtypes.push(ValType::Ref(RefType {
-                    nullable,
-                    // TODO: handle shared
-                    heap_type: HeapType::Abstract { shared: false, ty },
-                }));
+            let shareability: &[bool] = if config.shared_everything_threads_enabled {
+                &[true, false]
+            } else {
+                &[false]
+            };
+            for &shared in shareability {
+                use AbstractHeapType::*;
+                for ty in [
+                    Any, Eq, I31, Array, Struct, None, Func, NoFunc, Extern, NoExtern,
+                ] {
+                    valtypes.push(ValType::Ref(RefType {
+                        nullable,
+                        heap_type: HeapType::Abstract { shared, ty },
+                    }));
+                }
```

Comment:
This could be an `extend` and will generally be a little faster since the space can be reserved one time up front.